app/backend/src/core/controllers/appraiser/handlers/TokenTransfers.ts

```diff
@@ -50,21 +50,9 @@ export class TokenTransferHandler extends BaseHandler {
   private extractTransferEvents(logs: Log[]): TransferLogEvent[] {
     const transferEvents: TransferLogEvent[] = [];
 
-    const transferEventSigHash = ethers.id("Transfer(address,address,uint256)");
-
     for (const log of logs) {
-      if (log.topics[0] === transferEventSigHash) {
-        const contractAddress = ethers.getAddress(log.address);
-        const fromAddress = ethers.getAddress(`0x${log.topics[1].slice(26)}`);
-        const toAddress = ethers.getAddress(`0x${log.topics[2].slice(26)}`);
-        const rawAmount = log.data === "0x" ? BigInt(0) : BigInt(log.data);
-        transferEvents.push({
-          fromAddress,
-          toAddress,
-          contractAddress,
-          rawAmount,
-        });
-      }
+      const transferEvent = this.provider.getTransferEvent(log);
```

Comment:
Tried my best to keep `Transfer`-parsing code common for all chains, but there are just too many differences for Starknet. As such, it is not chain dependent and moved to `IBlockchainClient`.